sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -60,44 +65,100 @@
      *
      * @param testName Name of the test session record.
      * @param testMode The {@link TestMode} for this interceptor.
-     * @throws IOException If {@code testMode} is {@link TestMode#PLAYBACK} and an existing test session record could
-     * not be located or the data could not be deserialized into an instance of {@link RecordedData}.
+     * @throws UncheckedIOException If {@code testMode} is {@link TestMode#PLAYBACK} and an existing test session record
+     * could not be located or the data could not be deserialized into an instance of {@link RecordedData}.
      * @throws NullPointerException If {@code testName} is {@code null}.
+     * @deprecated Use {@link #InterceptorManager(String, TestMode, boolean)} instead.
      */
-    public InterceptorManager(String testName, TestMode testMode) throws IOException {
+    @Deprecated
```

Comment:
Don't need to deprecate this. `false` for `doNotRecord` is a good default.

Reply:
I was thinking this morning whether the new overload should take this boolean or a TestContextManager object. The former is more direct on the change but the latter will allow extensibility in the future without changing the API, thoughts?